src/AttachAddon.ts

```diff
@@ -14,96 +15,50 @@ export interface ITerminalAddon {
 }
 
 export class AttachAddon implements ITerminalAddon {
-  private _socket: WebSocket;
-  private _textDecoder: TextDecoder = new TextDecoder();
-  private _buffered: boolean;
-  private _attachSocketBuffer: string;
+  private _bidirectional: boolean;
+  private _utf8: boolean;
   private _disposables: IDisposable[] = [];
-  private _terminal: Terminal;
   private _dataListener: (data: string) => void;
 
-  public activate(terminal: Terminal): void {
-    this._terminal = terminal;
-  }
-
-  public dispose(): void {
-    this._disposables.forEach(d => d.dispose());
-    this._socket = null;
-  }
-
-  public attach(socket: WebSocket, bidirectional?: boolean, buffered?: boolean): void {
-    bidirectional = (typeof bidirectional === 'undefined') ? true : bidirectional;
-    this._socket = socket;
-    this._buffered = buffered;
-
-    this._disposables.push(addSocketListener(socket, 'message', (ev: MessageEvent) => this._getMessage(ev)));
-
-    if (bidirectional) {
-      this._dataListener = data => this._sendData(data);
-      this._disposables.push(this._terminal.addDisposableListener('data', this._dataListener));
+  constructor(public socket: WebSocket, options?: IAttachOptions) {
+    this._bidirectional = (options && options.bidirectional === false) ? false : true;
+    this._utf8 = options && options.inputUtf8;
+    if (this._utf8) {
+      this.socket.binaryType = 'arraybuffer';
     }
-
-    this._disposables.push(addSocketListener(socket, 'close', () => this.dispose()));
-    this._disposables.push(addSocketListener(socket, 'error', () => this.dispose()));
-  }
-
-  private _flushBuffer(): void {
-    this._terminal.write(this._attachSocketBuffer);
-    this._attachSocketBuffer = null;
   }
 
-  private _pushToBuffer(data: string): void {
-    if (this._attachSocketBuffer) {
-      this._attachSocketBuffer += data;
+  public activate(terminal: Terminal): void {
+    if (this._utf8) {
+      this.socket.binaryType = 'arraybuffer';
+      this._disposables.push(addSocketListener(this.socket, 'message',
+        (ev: MessageEvent) => (terminal as any).writeUtf8(new Uint8Array(ev.data))));
     } else {
-      this._attachSocketBuffer = data;
-      setTimeout(this._flushBuffer, 10);
+      this._disposables.push(addSocketListener(this.socket, 'message',
+        (ev: MessageEvent) => (terminal as any).write(ev.data)));
```

Comment:
This is the safer option if you rely on new API:

https://github.com/xtermjs/xterm-addon-search/blob/a7b579c408e03e836493341784827531aa2871ec/src/SearchAddon.ts#L26-L31

Reply:
Ah ok.